ksqldb-api-client/src/test/java/io/confluent/ksql/api/client/integration/ClientIntegrationTest.java

```diff
@@ -442,6 +510,36 @@ public void shouldStreamPullQueryOnEmptyStreamSync() throws Exception {
     assertThatEventually(streamedQueryResult::isComplete, is(true));
   }
 
+  @Test
+  public void shouldStreamPullQueryOnTruncatedStreamSync() throws Exception {
+    // double-check to make sure it's really truncated
+    truncateTopic(TRUNCATED_TEST_TOPIC);
+
+    // When
+    final StreamedQueryResult streamedQueryResult = client.streamQuery(
+        "SELECT * FROM " + TRUNCATED_TEST_STREAM + ";").get();
+
+    // Then
+    assertThat(streamedQueryResult.columnNames(), is(TEST_COLUMN_NAMES));
+    assertThat(streamedQueryResult.columnTypes(), is(TEST_COLUMN_TYPES));
+    assertThat(streamedQueryResult.queryID(), is(notNullValue()));
+
+    final List<Row> results = new LinkedList<>();
+    Row row;
+    while (true) {
+      row = streamedQueryResult.poll();
```

Comment:
Just wondering in normal scenarios, could it be possible that the first few polls return empty as the operator is not yet ready to fetch the first record? In other words do we guarantee that the first poll will always return some data as long as they are available?

Reply:
I'm not sure I follow. When you start the query, it resets its offsets to "earliest" and checks for the "end offsets". Then it reads from the start point to the end point. IIUC, those "start" and "end" offset admin client checks will hit the leaders of the partitions, so if there was an acknowledged send to the topic, then a subsequent query would see the sent record.

Reply:
What I was asking is in the `client.streamQuery` call are we guaranteed that the first `poll()` would always return the data, since if the fetching is async then maybe it does not guarantee --- similar to `consumer.poll()` which may not guarantee that the first `poll()` would always return data since the fetching IO is async. In the case, here if we beak immediately after the first `poll` returns no rows we may not be covering exactly what we want to test.

I cannot honestly remember how `streamQuery` is executed behind the scene, so if that's not an issue please ignore :P